plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -188,6 +192,50 @@ public FileHiveMetastore(NodeVersion nodeVersion, HdfsEnvironment hdfsEnvironmen
         }
     }
 
+    @Override
+    public synchronized MetastoreLock acquireTableExclusiveLock(
+            HiveIdentity identity,
+            AcidTransactionOwner transactionOwner,
+            String queryId,
+            String dbName,
+            String tableName)
+    {
+        SchemaTableName schemaTableName = new SchemaTableName(dbName, tableName);
+
+        ReentrantLock lock = tableLocks.computeIfAbsent(schemaTableName, ignored -> new ReentrantLock());
+        lock.lock();
+
+        tableLocks.put(schemaTableName, lock);
+        return new FileMetastoreLock(schemaTableName);
+    }
+
+    private class FileMetastoreLock
+            implements MetastoreLock
+    {
+        private final SchemaTableName tableName;
+        boolean released;
+
+        public FileMetastoreLock(SchemaTableName tableName)
+        {
+            this.tableName = requireNonNull(tableName, "tableName is null");
+        }
+
+        @Override
+        public void release()
+        {
+            if (released) {
+                throw new IllegalStateException("Table lock has already been released");
+            }
+            released = true;
+            ReentrantLock lock = tableLocks.get(tableName);
+            // Currently, metastore locks are always acquired and released in the same thread.
+            lock.unlock();
+            if (!lock.isLocked()) {
+                tableLocks.remove(tableName);
```

Comment:
`remove(tableName, lock)` to ensure you remove the thing you want to

Reply:
more importantly, i don't think we can base our logic on `lock.isLocked`.
there can be a thread doing acquire which pulled a lock from the map but didn't `.lock()` on it yet.
it would not be correct to remove that from the map.

i think we can do 
- replace Lock with some other primitive (like `boolean locked`) and implement locking based on the fact the class is synchronized
- implement locking with CDL or Sempahore to allow locks cleanup 

however, the more i am thinking about this the more i feel we shouldn't do this here. it's just distracting from the main PR.

let's remove removing from the map. and add a TODO comment here.
I don't think such leak can be a problem in tests and for more sustained use this will need to be revisited.

```
// There is a memory leak. TODO remove unused locks from tableLocks.
// Currently, there is no cleanup as this is used for testing purposes where the whole metastore instance 
// is not longed-lived.
```